src/typeEvaluator/typeEvaluate.ts

```diff
@@ -376,13 +376,31 @@ function handleFlatMap(node: FlatMapNode, scope: Scope): TypeNode {
   return mapArray(base, scope, (base) => {
     const inner = walk({node: node.expr, scope: scope.createHidden([base.of])})
 
-    return mapConcrete(inner, scope, (inner) => {
-      if (inner.type === 'array') {
-        return inner
-      }
+    return mapConcrete(
+      inner,
+      scope,
+      (inner) => {
+        if (inner.type === 'array') {
+          return inner
+        }
 
-      return {type: 'array', of: inner}
-    })
+        return {type: 'array', of: inner}
+      },
+      (nodes) => {
+        const inner: TypeNode[] = []
+        for (const node of nodes) {
+          // Bail out early if we've detected an unknown.
+          if (node.type === 'unknown') return {type: 'array', of: node}
```

Comment:
I _think_ this is fine, but why do we want to bail out early? 

We are currently missing some parts of the groq spec that we know will return `unknown` ie a `count(..)` so if you are flat mapping a list thats based on count it will short circuit the whole thing.

Reply:
I'm actually not sure if we ever end up generating a union of `unknown` with another type (e.g. `unknown | string`). And even if so, I'm not sure what value it is for FlatMap to then return `Array<unknown | string>` 🤔

Reply:
I think this is such an edge case, and even then we do return `unknown` so leaving it up the caller to handle it - so fine to merge at least 😅